### PR TITLE
google: add compute instance

### PR DESCRIPTION
**Pull request by tormath1**
This PR is based on the Google boilerplate. I import Google Compute Instances using this [api](https://cloud.google.com/compute/docs/reference/rest/v1/instances/list).

TC been tested against 2 instances: in different zones and different labels.

What I've done: 

- isolate `requiredStringFlags` function into `root.go` since we are using the same function in both command lines
- fetch `zone` list since the API need both `project` and `zone` to list instances. We fetch zones from the given `region`
- implement a simple `reader` in waiting for RAWS removal
- ~hack the ID between read and import~

I'll squash commits before merging.

### CODE REVIEW

File: cmd/google.go [hunk 1]
```diff
@@ -14,12 +26,83 @@ var (
...
+
+	googleCmd.Flags().String("project", "", "project (required)")
+	_ = viper.BindPFlag("project", googleCmd.Flags().Lookup("project"))
+
+	googleCmd.Flags().String("region", "", "region (required)")
+	_ = viper.BindPFlag("region", googleCmd.Flags().Lookup("region"))
+
+	// Filter flags
+
+	googleCmd.Flags().StringSliceVarP(&tags, "tags", "t", []string{}, "List of tags to filter with format 'NAME:VALUE'")
```
Comment (xescugc) [inline cmd/google.go]: Does Google use the `tags` name? Isn't `labels` or something like that?

File: google/provider.go [hunk 2]
```diff
@@ -1 +1,56 @@
...
+	}, nil
+}
+
+func (g *google) HasResourceType(t string) bool {
+	return false
+}
+
+func (g *google) Region() string { return "region" }
+func (g *google) String() string { return "google" }
+func (g *google) TagKey() string { return "tags" }
```
Comment (xescugc) [inline google/provider.go]: Same, isn't Google using `labels`?

File: google/reader.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+)
+
+type GCPReader struct {
+	compute *compute.Service
+	project string
+	region  string
+	//TODO: add filters
```
Comment (xescugc) [inline google/reader.go]: If there is a TODO add an issue for it and add the link to the TODO.

File: google/reader.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	}, nil
+}
+
+func (r *GCPReader) getZones() ([]string, error) {
+	rs := compute.NewRegionsService(r.compute)
+	region, err := rs.Get(r.project, r.region).Do()
+	if err != nil {
+		return nil, err
+	}
+	// zones are URL format. Need to split them
```
Comment (xescugc) [inline google/reader.go]: Can you put an example? I see for the code what it is (it's the last element of the URL that we want) but just to have itmore clear.

File: google/reader.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	zones := make([]string, 0)
```
Comment (xescugc) [inline google/reader.go]: `zones := make([]stirng,0, len(region.Zones))` so you already allocate the needed space.

File: google/reader.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+		tmp := strings.Split(URL, "/")
+		zones = append(zones, tmp[len(tmp)-1])
+	}
+	return zones, nil
+}
+
+// ListInstances return a list of instances in the project/region
+func (r *GCPReader) ListInstances() (map[string][]compute.Instance, error) {
+	is := compute.NewInstancesService(r.compute)
+	//TODO: add filters
```
Comment (xescugc) [inline google/reader.go]: Same, add issue if this is relevant and link it.

File: google/reader.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+	instancesList := make(map[string][]compute.Instance, 0)
+	zones, err := r.getZones()
+	if err != nil {
+		return nil, err
+	}
+	for _, zone := range zones {
+		instances := make([]compute.Instance, 0)
+		il, err := is.List(r.project, zone).Do()
```
Comment (xescugc) [inline google/reader.go]: Dest this return all of them? or it's paginated?

File: google/reader.go [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+	return &GCPReader{
+		compute: comp,
+		project: project,
+		region:  region,
+	}, nil
+}
+
+func (r *GCPReader) getZones() ([]string, error) {
+	rs := compute.NewRegionsService(r.compute)
+	region, err := rs.Get(r.project, r.region).Do()
```
Comment (xescugc) [inline google/reader.go]: Does this return all of them or paginated?

Comment (tormath1) [reply google/reader.go]: it returns actually only one region.

Comment (xescugc) [reply google/reader.go]: Cool, IDK why I thought it would return multiple

File: google/reader.go [hunk 9]
```diff
@@ -54,16 +54,18 @@ func (r *GCPReader) ListInstances(filter string) (map[string][]compute.Instance,
 		return nil, err
 	}
 	for _, zone := range zones {
-		instances := make([]compute.Instance, 0)
-		il, err := is.List(r.project, zone).
+		// 500 is the current `maxResults`
+		instances := make([]compute.Instance, 0, 500)
```
Comment (xescugc) [inline google/reader.go]: And if you have more than 500? It's not possible to request them all?

Comment (tormath1) [reply google/reader.go]: I use pages. Each page has a default size of 500 elements: see [pagination](https://godoc.org/google.golang.org/api/compute/v1#InstancesListCall.Pages) and [max result](https://godoc.org/google.golang.org/api/compute/v1#InstancesListCall.MaxResults)

`instances` slice will store only the instances returned by the current page (so 500 maximum). Then we append this slice to the final `instancesList` map.

**Comment by tormath1** [comment]
ready for a next review :)

**Comment by tormath1** [comment]
rebased and squashed :heavy_check_mark:

**Pull request merged**